new/vcloud-manager-blueprint.yaml

```diff
@@ -0,0 +1,867 @@
+
+  manager_host:
+    type: cloudify.nodes.ManagerHost
+    relationships:
+      - type: cloudify.relationships.contained_in
+        target: manager_host_server
+    properties:
+      install_agent: false
+      ip: { get_attribute: [manager_host_server, ip] }
+      public_ip: { get_attribute: [manager_public_ip, public_ip] }
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: fabric.fabric_plugin.tasks.run_script
+          inputs:
+            script_path: components/vcloud/update-vcloud-vm.sh
+            fabric_env:
+              user: { get_input: ssh_user }
+              key_filename: { get_input: ssh_key_filename }
+              host_string: { get_attribute: [manager_public_ip, public_ip] }
+              port: { get_input: ssh_port }
+              disable_known_hosts: True
+        configure:
+          implementation: fabric.fabric_plugin.tasks.run_task
+          inputs:
+            tasks_file: components/vcloud/configure.py
+            task_name: configure
+            task_properties:
+              vcloud_config: { get_property: [vcloud_configuration, vcloud_config] }
+            fabric_env:
+              user: { get_input: ssh_user }
+              key_filename: { get_input: ssh_key_filename }
+              host_string: { get_attribute: [manager_public_ip, public_ip] }
+              port: { get_input: ssh_port }
+              disable_known_hosts: True
+
+  # #####################################################################
+  # The manager_configuration node is meant to be read by Cloudify to
+  # provide runtime configuration and information for the CLI and the
+  # Manager.
+  # #####################################################################
+  manager_configuration:
+    type: cloudify.nodes.MyCloudifyManager
+    properties:
+      ssh_user: { get_input: ssh_user }
+      ssh_key_filename: { get_input: ssh_key_filename }
+      agent_local_key_path: { get_input: agent_private_key_path }
+      cloudify:
+        resources_prefix: { get_input: resources_prefix }
+
+        transient_deployment_workers_mode:
+          enabled: false
+          global_parallel_executions_limit: 50
+
+        cloudify_agent:
+          min_workers: 0
+          max_workers: 5
+          remote_execution_port: 22
+          user: { get_input: agents_user }
+
+        workflows:
+          task_retries: -1  # this means forever
+          task_retry_interval: 30
+
+        policy_engine:
+          start_timeout: 30
+
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: fabric.fabric_plugin.tasks.run_script
+          inputs:
+            script_path: components/manager/scripts/configure_manager.sh
+            fabric_env:
+              user: { get_input: ssh_user }
+              key_filename: { get_input: ssh_key_filename }
+              host_string: { get_attribute: [manager_public_ip, public_ip] }
```

Comment:
I see that in the manager_host node you also provide an ssh port to the fabric_env. As this node relates directly to the manager_host, shouldn't we apply the same port here?
 
Moreover, you can use yaml aliases instead of redefining the fabric env dict.

```yaml
fabric_env: &my_fabric_env
   host_string: ...
   ... 

...
fabric_env: *my_fabric_env
```